### DISCUSSION TITLE
Sidebar Tabs

### DISCUSSION THREAD

**Discussion by ernst86913** [directus/directus#5459] [Feature Requests]
@benhaynes 

As the sidebar gets overly crowded, and also as not all information is of equal priority, I suggest to introduce sidebar sliding tabs to have more structure and free up space in this sidebar.

Attached you can find a working prototype which I just created... still needs some tweaking styling... but it should give you some idea.

a) Is this something you are currently working on?
b) Is this something you think makes sense for your product?

I will go ahead and will develop it for my purposes... if you are interested let me know and I will make sure to do it in a more productized way.

PS - obviously the content of the tabs should be different and not always the same... and... there can either be just 1 tab-slot in which case the tabs will be hidden... or max to 4 tabs. I would also integrate the close sidebar button into the top bar if tabs are used (= more than 1 tab)

KR
Roman

https://user-images.githubusercontent.com/47870436/117039819-81dd3180-ad09-11eb-8722-19695c5ddfa8.mp4

**Comment by scouech** [comment]
Hey! UX Designer here.

I've been playing with Directus for a while now and I see what you are trying to achieve with your example but for accessibility and on small screens it's not super ideal 🤓
I wonder if the entire right sidebar system could be changed for something more similar to table headers like on Notion for example?
(if you don't see what I mean I could come up with something)

**Reply by ernst86913** [reply]
Sure - open for any suggestions... important is that @benhaynes @rijkvanzanten want to go in this direction

**Reply by rijkvanzanten** [reply]
I agree with @scouech here. Not sure if the benefit of a less cluttered sidebar is worth the additional UX downside of the limited discoverability of hiding them away under those tabs 🤔 (Also, I wouldn't really know what those tabs would be / how to group the sections)

**Reply by benhaynes** [reply]
I like the effect, and this is a clean and considered approach. Is this meant to complement/organize the existing sidebar components, or replace them? I agree with @scouech and @rijkvanzanten — Going horizontal here will create an artificial limit in items, and removes the text/title. Also, it seems like it might be a bit jarring to transition from the sidebar closed state to the open one (right now things stay in the same place).

This might be a good way to organize the sidebar component if we end up with a few more and the length becomes too much. I could see splitting the components into categories. But for now, I don't think we have enough in that sidebar to warrant this added complexity.

Love the design and PoC though. Let me know if you're looking for any tasks to work on, we can always use the help building new interfaces!

**Reply by ernst86913** [reply]
@benhaynes 

Written in a way to complement the existing sidebar... if you don't add tabs as input slots, the sidebar shows as it shows right now. It's not about going completely horizontal... but introducing meaningful groups. And still have the mechanisms which you are having right now for each of the groups

The issues I see with the current sidebar:
a) we are mixing items which are used regularly with items which are used maybe just once during setup... eg... who wants to read the "info" every time... or the "refresh settings"
b) on "laptop resolutions" the group items e.g. in collection layout meanwhile consume all the space... opening one of them requires you to scroll long time in a very small scroll area.
c) Example for grouping:  MAIN, ACTIVITY, SETTINGS, INFO  -  MAIN would be items which might be used every time opening an item/collection... like comments, filters,...   ACTIVITY would list all activity that happened for a particular item... could allow a user to enable user notifications for an item using a button... and also house the revisions; SETTINGS for configuration options which are more of one-time type (e.g. refresh, layout,....); INFO would be for more detailed in-app help page... with real space instead of squeezing things

On mobile you can use it as is by sliding the whole sidebar in...

As mentioned... the current implementation also is reaching its limits... some actions like "Export" could be shifted to an action header button... but knowing some implementations of large vendors (e.g. ServiceNow) they all end up with this kind of design.

Will implement that one for my personal use...   sure... let me know if I can help somewhere... can't promise any delivery dates as I have quite some work to do in parallel... but I like directus and its a nice hobby for me.

**Reply by rijkvanzanten** [reply]
> we are mixing items which are used regularly with items which are used maybe just once during setup

This is a bit of an opinionated point though 🤔 Some people might not care about "auto refresh" or its configuration at all, while others use it and configure it very often. The same goes for layout options, and basically any of the other sidebar sections.

**Reply by ernst86913** [reply]
BTW... some addtl. thoughts... the bookmarks in Directus are actually not bookmarks linking to items, but they are actually VIEWS on collections... I would move those to the right sidebar as well and remove them from the main menu.

The notifications on the right sidebar should go to the left main menu... at least if its about user notifications where a user needs to action on them. Notifications are not item / collection specific (at least thats how I would understand them) and therefore don't belong to the right sidebar... confuses me a lot...

**Reply by ernst86913** [reply]
@rijkvanzanten I am thinking about the 200 standard users in a company... vs the 2 guys in IT configuring it.... those 200 standard users rely on the 2 guys to make it right.... Regarding layout options... as standard user in a company I would configure those once and save them as views / bookmarks... and always use those. BTW... as one of the 2 guys in the company I would actually think about restricting the other 200 users to those options.... as it will heavily increase the maintenance... but - I agree... its an opinionated point of view based on my experience...

**Reply by ernst86913** [reply]
... and as always depends on the use case.

**Comment by benhaynes** [comment]
Thanks, @ernst86913 — I'll respond to the Sidebar Tabs here, though I think the other points (Notifications, Bookmarks, etc) should have dedicated Discussions so things don't get muddled here. Responses to your points...

A. As Rijk said, this can be a bit subjective... and might change over time. The `Info` component will hold more _dynamic_ info about the page in time (counts, details, etc), similar to how it works on the File Detail page now. So that is useful more than once. Auto-Refresh and other "config" options might be updated by some users more than others. Also, keep in mind that the true Layout configuration is done by admins within Settings > Presets... not on the Layout sidebar (which is for user overrides).

B. This can be solved by vertically smooth scrolling to the sidebar component you opened (when there isn't enough space in the viewport for everything)... and maybe even scrolling past the other component headers (so they're only sticky in desktop mode). With this, the current design would actually offer _more_ room... assuming your top tabs are fixed in the sidebar.

C. I think I get the organization you're hinting at here, but that might be pretty specific to your viewport size (which is relevant). On my screen (see below), those three Item Detail components don't take up a lot of room at all... or demand more organizational structure. Also, I'm curious how your design would look in the collapsed state. One thing I really like about what we have now is that everything is only one click away from being open... and having those icons always visible also means we can show badges/count/info to the user without opening/clicking.

![Screen Shot 2021-05-05 at 3 34 38 PM](https://user-images.githubusercontent.com/522079/117199084-c58e7480-adb7-11eb-8411-d4a59c10c2d4.png)

I'm not completely against this direction, I just don't think we're at a point yet where this is required... and so I'd rather delay this change until it's more relevant (and we have a better sense of the components (count/type/content) will be on various pages. Also, one _big_ change that will be happening will be opening these info sidebar components up as a new extension type. That seems like a more appropriate time to look into a broader organizational structure here.

As with all Feature Requests, I think it's best to also get more community feedback on this... so we know all the different angles/use-cases! I'd love to see a few more 👍  and ⬆️  on here before making any big adjustments.

**Reply by ernst86913** [reply]
Thanks for taking the time to respond in that detail. Very much appreciated

**Reply by ernst86913** [reply]
@benhaynes "Also, one big change that will be happening will be opening these info sidebar components up as a new extension type."

That would definitely be a great functionality... Same needs to be done with the action buttons I assume....

**Reply by benhaynes** [reply]
> Same needs to be done with the action buttons I assume

Potentially. Action buttons are a bit more "system"... and platform critical. But we recognize that it's a desired feature.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
